Source/Headspring.BulkWriter/Mapping.cs

```diff
@@ -43,9 +41,15 @@ public IBulkWriter<TResult> CreateBulkWriter(string connectionString)
             SqlBulkCopyOptions sqlBulkCopyOptions = hasAnyKeys ? SqlBulkCopyOptions.KeepIdentity : SqlBulkCopyOptions.Default;
             var sqlBulkCopy = new SqlBulkCopy(connectionString, sqlBulkCopyOptions)
             {
-                DestinationTableName = this.destinationTableName
+                DestinationTableName = this.destinationTableName,
+                BatchSize = options.BatchSize,
+                BulkCopyTimeout = options.BulkCopyTimeout,
+                EnableStreaming = options.EnableStreaming,
+                NotifyAfter = options.NotifyAfter,
             };
 
+            sqlBulkCopy.SqlRowsCopied += options.SqlRowsCopied;
```

Comment:
This may be more a preference than anything -- so you can ignore it if it doesn't makes sense to you. To me, it would make more sense for this event and the properties on BulkWriterOptions to be declared on the IBulkWriter<T> interface, rather than on the specialized options class. At the very least, by taking this approach, you wouldn't need to add the additional method on the IMapping<T> interface that accepts the options class.